lib/ApplicationFeatures/V8SecurityFeature.cpp

```
@@ -369,6 +371,16 @@ bool V8SecurityFeature::shouldExposeEnvironmentVariable(v8::Isolate* isolate,
                                 _environmentVariablesBlacklistRegex);
 }
 
+bool V8SecurityFeature::isAllowedToAccessSystemDatabase(v8::Isolate* isolate) const {
+  TRI_GET_GLOBALS();
+  TRI_ASSERT(v8g != nullptr);
+  auto const& sec = v8g->_securityContext;
+  auto vocbase = GetContextVocBasePointer(isolate);
+  bool isSystemDataBase = !vocbase || ( vocbase->name() == "_system" ); // If the vocbase is a nullptr we are in arangosh
```

can we have a fail early if-operation here?